api/citations/utils.py

```diff
@@ -75,4 +110,148 @@ def render_citation(node, style='apa'):
     elif cit.count(title) == 0:
         cit = clean_up_common_errors(cit)
 
+    if isinstance(node, PreprintService):
+        cit_node = node.node
+    else:
+        cit_node = node
+
+    if style == 'apa':
+        cit = apa_reformat(cit_node, cit)
+    if style == 'chicago-author-date':
+        cit = chicago_reformat(cit_node, cit)
+    if style == 'modern-language-association':
+        cit = mla_reformat(cit_node, cit)
+
+    return cit
+
+
+def apa_reformat(node, cit):
+    new_csl = cit.split('(')
+    contributors_list = [x for x in node.contributors if node.get_visible(x)]
+
+    # throw error if there is no visible contributor
+    if len(contributors_list) == 0:
```

Comment:
Instead of checking `len(contributors_list)` several times in this function, assign the length to a variable and check that.